conda/core/solve.py

```diff
@@ -266,6 +273,53 @@ def solve_final_state(self, update_modifier=NULL, deps_modifier=NULL, prune=NULL
 
         return ssc.solution_precs
 
+    def get_request_package_in_solution(self, solution_precs, specs_map):
+        requested_packages = {}
+        for pkg in self.specs_to_add:
+            update_pkg_request = pkg.name
+
+            requested_packages[update_pkg_request] = [
+                (i.name, str(i.version)) for i in solution_precs
+                if i.name == update_pkg_request and i.version is not None
+            ]
+            requested_packages[update_pkg_request].extend(
+                [(v.name, str(v.version)) for k, v in specs_map.items()
+                 if k == update_pkg_request and v.version is not None])
+
+        return requested_packages
+
+    def packages_in_solution_change(self, pre_packages, post_packages, index_keys):
+        update_constrained = False
+
+        def empty_package_list(pkg):
+            for k, v in pkg.items():
+                if len(v) == 0:
+                    return True
+            return False
+
+        if empty_package_list(pre_packages) or empty_package_list(post_packages):
+            return update_constrained
+
+        for pkg in self.specs_to_add:
+            current_version = max(i[1] for i in pre_packages[pkg.name])
+            if current_version == max(i.version for i in index_keys if i.name == pkg.name):
+                continue
+            else:
+                if post_packages == pre_packages:
+                    for pkg_name, matches in post_packages.items():
+                        pkg_match = [m for m in matches if m[1] is not None][0]
+                        print('\n\nTrying to update package {update_pkg} however it seems like '
+                              'you already have this package constrained to \n    {const_pkg} '
```

Comment:
This is a great start.  I think one critical detail is missing, though.  You report a version that is the result of one or more constraints, but the thing we really need most is to understand the constraint itself, and where it comes from.  In order to do this, you'll need to look through all of the specs that factored into the solution, and find which ones have constraints (directly, or perhaps buried deeper in the dependency tree) that affect this package.